pkg/services/provisioning/dashboards/file_reader.go

```diff
@@ -78,6 +78,7 @@ func (fr *fileReader) ReadAndListen(ctx context.Context) error {
 	}
 }
 
+// startWalkingDisk finds and saves dashboards on disk.
```

Comment:
In this function we have `sanityChecker` code that checks for duplicate UIDs in dashboards. I think it would be good idea to have that also for folders. It needs to be lifted though probably into `configReader.readConfig()` function where it would have access to all the configs and folders UIDs

Reply:
Yes that makes sense. Detecting duplicate UID's can save operators many hours.